torch/csrc/jit/codegen/cuda/fusion.cpp

```diff
@@ -541,6 +549,39 @@ bool Fusion::hasGridReduction() {
   return false;
 }
 
+std::vector<Val*> Fusion::getTerminatingOutputs() {
+  FusionGuard fg(this);
+
+  std::unordered_set<Val*> used_vals;
+
+  const auto exprs = ExprSort::getExprs(
+      this, std::vector<Val*>(outputs().begin(), outputs().end()));
+
+  for (auto expr : exprs) {
+    for (auto inp : expr->inputs())
+      used_vals.emplace(inp);
+  }
+
+  std::unordered_set<Val*> terminating_outputs;
```

Comment:
I understand I'm criticizing my own code, but to get deterministic ordering, couldn't we simply use the order of `outputs()`? I'm uncertain why I didn't do:
```
std::vector<Val*> terminating_outputs;
  for (auto out : outputs()) {
    if (used_vals.find(out) != used_vals.end())
      continue;
    terminating_outputs.push_back(out);
  }
```
Wouldn't this also give us run to run determinism?

Reply:
That's right and much better! Change applied.